src/liquid/components/ld-notification/ld-notification.tsx

```diff
@@ -0,0 +1,190 @@
+      return
+    }
+
+    const firstErrorNotificationIndex = this.queue.findIndex(
+      (notification) => notification.type === 'alert'
+    )
+    if (firstErrorNotificationIndex === -1) {
+      this.queue = [...this.queue, newNotification]
+      this.currentNotification = newNotification
+      return
+    }
+
+    this.queue.splice(firstErrorNotificationIndex, 0, newNotification)
+    this.queue = [...this.queue]
+  }
+
+  @Listen('ldNotificationDismiss', {
+    target: 'window',
+    passive: true,
+  })
+  handleNotificationDismiss() {
+    if (!this.currentNotification) return
+
+    this.queueDismissed.unshift(this.queue.pop())
+    this.queueDismissed = [...this.queueDismissed]
+    this.queue = [...this.queue]
+    this.currentNotification = this.queue[this.queue.length - 1]
+
+    this.fadeoutTimeouts.push(
+      setTimeout(() => {
+        this.queueDismissed = this.queueDismissed.slice(0, -1)
+      }, FADE_TRANSITION_DURATION)
+    )
+  }
+
+  @Listen('ldNotificationClear', {
+    target: 'window',
+    passive: true,
+  })
+  handleNotificationClear() {
+    this.queueDismissed = [...this.queue]
+    this.queue = []
+    this.currentNotification = undefined
+    this.fadeoutTimeouts.forEach((timeoutId) => clearTimeout(timeoutId))
+    this.fadeoutTimeouts.push(
+      setTimeout(() => {
+        this.queueDismissed = []
+      }, FADE_TRANSITION_DURATION)
+    )
+  }
+
+  disconnectedCallback() {
+    clearTimeout(this.dismissTimeout)
+    this.fadeoutTimeouts.forEach((timeoutId) => clearTimeout(timeoutId))
+  }
+
+  render() {
+    const dismissIcon = `
+      <svg
+        width="24"
+        height="24"
+        viewBox="0 0 24 24"
+        fill="none"
+        xmlns="http://www.w3.org/2000/svg"
+      >
+        <title>Dismiss</title>
+        <path
+          d="M6 6L18 18"
+          stroke="currentColor"
+          stroke-width="3"
+          stroke-linecap="round"
+          stroke-linejoin="round"
+        />
+        <path
+          d="M6 18L18 6"
+          stroke="currentColor"
+          stroke-width="3"
+          stroke-linecap="round"
+          stroke-linejoin="round"
+        />
+      </svg>
+    `
+
+    let cl = 'ld-notification'
+    cl += ` ld-notification--${this.placement}`
+
+    return (
+      <Host class={cl} role="region" aria-label="Notifications">
+        {this.queue.map((notification, index) => (
+          <div
+            class={`ld-notification__item ld-notification__item--${notification.type}`}
+            key={index}
+          >
+            <div
+              class="ld-notification__item-content"
+              innerHTML={notification.content}
+              role={notification.type === 'alert' ? 'alert' : 'status'}
+            ></div>
+            <button
+              class="ld-notification__btn-dismiss"
+              onClick={this.handleNotificationDismiss.bind(this)}
+              innerHTML={dismissIcon}
+            ></button>
+          </div>
+        ))}
```

Comment:
I don't know. I like it the way it is 😄  Otherwise I would need to scroll to the render runction in order to see what's in there... But this is a taste thing I guess.

Reply:
I see. To keep it DRY one could also write something like this:

```js
        {this.queue.concat(this.queueDismissed).map((notification) => (
          <div
            class={`ld-notification__item${notification.dismissed ? ' ld-notification__item--dismissed' : ''} ld-notification__item--${notification.type}`}
            key={notification.type + notification.content}
          >
            <div
              class="ld-notification__item-content"
              innerHTML={notification.content}
              role={notification.type === 'alert' ? 'alert' : 'status'}
            ></div>
            <button
              class="ld-notification__btn-dismiss"
              onClick={this.handleNotificationDismiss.bind(this)}
            >
              {this.getDismissIcon()}
            </button>
          </div>
        ))}
```

Would either need another `map` or refactoring upfront to add the `dismissed` attribute inside queueDismissed. Putting time into this only makes sense if you agree on my other comments on the attributes of dismissed elements, though.